etc/submit.sh

```diff
@@ -10,37 +10,100 @@ sed -e 's/^/export /' startup_environment.sh > tmp_env.sh
 mv tmp_env.sh startup_environment.sh
 export JOBSTARTDIR=$PWD
 
+# Saving START_TIME and when job finishes END_TIME.
+# Multiple checks are done:
+# 1) Check submit.sh bootstrap exit code.
+#    If it is not 0 sleep MIN_JOB_RUNTIME_SECONDS and in case it is 0, force to sleep 5 mins total runtime
+#    If this bootstrap script is exiting non 0 exit code, something is wrong on this WN...
+# 2) If Job exited 0, do not sleep any timeout
+# 3) Otherwise, sleep up to MIN_JOB_RUNTIME_TIME_SECONDS seconds of runtime.
+#    By default MIN_JOB_RUNTIME_TIME_SECONDS will be set to 0, if this is not passed through agent schedd configuration
+# Fast finishing jobs (failed specially) might overload schedulers and 
+# restarting jobs on same bad WN leads to more failures
+START_TIME=$(date +%s)
+function finish {
+  exitCode=$?
+  echo "======== WMAgent final job runtime checks STARTING at $(TZ=GMT date) on $(hostname) ========"
```

Comment:
use `hostname -f` instead please. I've seen cases where just `hostname` gives us the short name, not fqdn. Update all the others as well.